### PR TITLE
{cae}[foss/2021a] OpenFOAM v2106

### LINKED ISSUE #13276: Integrate OpenFOAM-v2106
[OpenFOAM v2106 was just released](https://www.openfoam.com/news/main-news/openfoam-v2106) - would be nice to have an EasyBuild pack to follow.

@boegel From what I can see, the config restructuring for the v2012 release should pay off. Likely only need to copy the `.eb` file with a new name and adjust the SHA values for the new tar file.

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)
Depends on:
- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/12977
- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/13269

edit (by @boegel): fixes #13276

### CODE REVIEW

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2106-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+toolchainopts = {'cstd': 'c++11'}
+
+source_urls = ['https://sourceforge.net/projects/openfoam/files/%(version)s/']
+sources = [SOURCE_TGZ]
+patches = [
+    ('OpenFOAM-v2012-cleanup.patch', 1),
+    'OpenFOAM-v1906-wmake-ompi.patch',
+]
+checksums = [
+    'e765881451d8d15ce33caa28850cdcc4534b8c4590e08ee6011bcafd11216940',  # OpenFOAM-v2106.tgz
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2106-foss-2021a.eb:23]: I'm getting a different SHA256 checksum for `OpenFOAM-v2106.tgz`: `11e41e5b9a253ef592a8f6b79f6aded623b28308192d02cec1327078523b5a37` 

Was there a sneaky re-release of OpenFOAM v2106 without a proper version bump? (@olesenm?!)

@robert-mijakovic Do you still have the original source tarball you were using, with checksum `e765881451d8d15ce33caa28850cdcc4534b8c4590e08ee6011bcafd11216940`?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2106-foss-2021a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2106-foss-2021a.eb:23]: The SHA256 in the Spack packages agrees with what I'm seeing, so I'll change this...

https://github.com/spack/spack/blob/e911f8ab3cfc073f77ea874871a95cb63dd928f4/var/spack/repos/builtin/packages/openfoam/package.py#L268

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3111.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/403541abec92c401c21a40fa29feb692 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
First compiler error from the build log of the failing build on `generoso` (CentOS 8):
```
/users/boegelbot/CentOS8/haswell/software/CGAL/5.2.2-gompi-2021a/include/CGAL/array.h:72:44: error: std::index_sequence has not been declared
   72 | make_filled_array_aux(const T& value, std::index_sequence<Is...>)
      |                                            ^~~~~~~~~~~~~~
```

This looks a lot like https://github.com/CGAL/cgal/issues/4974, but that was fixed in https://github.com/CGAL/cgal/pull/4981, which is included in CGAL 5.2.2.

I'm puzzled why this problem only pops up on `generoso`, and not on our CentOS 7 Intel Skylake cluster (see other test report)... It doesn't like like this is caused by a newer glibc?!

@olsenm Is OpenFOAM v2106 supposed to be compatible yet with CGAL 5.x? For v2012, we were still using CGAL 4.14.3...

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
I've swapped to CGAL 4.14.3 (latest 4.x) as dependency for OpenFOAM v2106, since that seems to work just fine...

I'll also use that same CGAL version for OpenFOAM 9 (even though we used CGAL 5.x there for OpenFOAM 8)

File: easybuild/easyconfigs/l/libcerf/libcerf-1.15-GCCcore-10.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+checksums = ['a5d45475e08d431267fd29d6af987a9dd9b6792578ec3feb466d4d21f2844868']
+
+builddependencies = [
+    ('binutils', '2.36.1'),
+    ('CMake', '3.20.1'),
+    ('Perl', '5.32.1'),  # required for pod2html
+]
+
+separate_build_dir = True
+
```
Comment (branfosj) [inline easybuild/easyconfigs/l/libcerf/libcerf-1.15-GCCcore-10.3.0.eb:31]:
```suggestion
```

Not needed in CMakeMake as it is the default.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2621.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/b36518d30fc14ad8ef6905524b4bf6cb for a full test report.

**Review by branfosj — APPROVED** [review]
lgtm

**Review by boegel — APPROVED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/cfe7562f107db14aa070c64189034780 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3502.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/8f175e7f47e235daa86a15137fcdb0d0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (3 easyconfigs in total)
node2621.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/362d543b3fcc6a62240e71b13ccc990f for a full test report.

**Pull request merged**